crates/bevy_input/src/gamepad.rs

```diff
@@ -2,6 +2,18 @@ use crate::{Axis, Input};
 use bevy_ecs::event::{EventReader, EventWriter};
 use bevy_ecs::system::{Res, ResMut, Resource};
 use bevy_utils::{tracing::info, HashMap, HashSet};
+use thiserror::Error;
+
+/// Errors that occur when setting settings for gamepad input
+#[derive(Error, Debug)]
+pub enum GamepadSettingsError {
+    #[error("{0}")]
+    InvalidAxisSetting(String),
```

Comment:
Storing a string in this error type doesn't really make sense to me. We should probably be storing the values set, and then providing doc strings on what the values mean.

Reply:
Redid all the `GamepadSettingsError`s to try to comply best with Rust API guidelines.